### ISSUE TITLE
[BUG] FATAL ERROR in filter when referencing variable that exists but has an undefined value

### ISSUE THREAD

**Issue by ericshulman** [Jermolene/TiddlyWiki5#7270]
Enter the following into a new tiddler:
```
<$let test1=<<test>> test2={{{ [<test1>] }}}>
```
Note that the `test` variable is undefined.  Result is a fatal error reporting:
"Uncaught Linked List only accepts string values, not undefined"

**Comment by saqimtiaz** [comment]
The underlying issues is that the $let widget does not specify a default value when looking up variables:

https://github.com/Jermolene/TiddlyWiki5/blob/925ce2b505bdd20352eb53a14dfac7033617e70a/core/modules/widgets/let.js#L77

Instead of something like this:
```
	options = options || {};
	options.defaultValue = "";
	return Widget.prototype.getVariableInfo.call(this,name,options);
```
I think it would make sense for an undefined variable to resolve to a blank string. Does that make sense @Jermolene ?

**Comment by Jermolene** [comment]
Thanks @saqimtiaz @ericshulman that's right, we've had some issues like this in the past. I'll push a fix shortly.

**Comment by Jermolene** [comment]
It's probably overkill, but I've used a pattern that avoids modifying the "options" object passed by the caller:

```js

	return Widget.prototype.getVariableInfo.call(this,name,$tw.utils.extend(Object.create(null),options,{
		defaultValue: ""
	}));
```

**Comment by Jermolene** [comment]
There were problems with the original fix (see #7348), and so a new fix in 589813f23335bbd3b50be52ad5d5211fbbffee33
